Library/Formula/bazel.rb

```diff
@@ -0,0 +1,25 @@
+class Bazel < Formula
+  homepage "http://bazel.io/"
+  url "https://github.com/google/bazel.git", :tag => "0.0.2"
+  sha256 "5865356e74622c248248174acd09f38c07f3983867d7e687fa5932f2216e84c0"
+
+  head "https://github.com/google/bazel.git"
+
+  depends_on "protobuf" => :build
+  depends_on "libarchive"
+
+  def install
+    if build.stable?
+      inreplace "compile.sh", "$(brew --prefix 2>/dev/null)/Cellar/libarchive/*/include/archive.h", "#{Formula["libarchive"].opt_include}/archive.h"
+    else
+      inreplace "compile.sh", "$(brew --prefix libarchive 2>/dev/null)/include/archive.h", "#{Formula["libarchive"].opt_include}/archive.h"
+    end
+
+    system "./compile.sh"
+    bin.install "./output/bazel"
+  end
+
+  test do
+    #TODO: tests; see https://github.com/Homebrew/homebrew/pull/38061#discussion-diff-27137134
```

Comment:
> You must run Bazel from within source code tree, which is properly configured for use with Bazel. We call such a tree a workspace directory. Bazel provides a default workspace directory with sample BUILD files and source code in base_workspace. The default workspace contains files and directories that must be present in order for Bazel to work. If you want to build from source outside the default workspace directory, copy the entire base_workspace directory to the new location before adding your BUILD and source files.

We could always steal the test they discuss there with a various degree of ` mkdir ` and ` (testpath/"horses/ducks").write << -EOS.undent `s and stage it in the testpath that way. The code involved doesn't look too long, and given the permissive license we could just 'borrow' it.